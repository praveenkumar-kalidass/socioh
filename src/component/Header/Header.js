import React from 'react';
import PropTypes from 'prop-types';
import { useWindowDimensions } from 'react-native';
import Svg, { Circle } from 'react-native-svg';

import { COLOR } from 'constant';
import {
  HeaderContainer,
  HeaderContent,
  HeaderContentContainer,
  Title,
} from './Header.style';

const Header = ({ scene }) => {
  const { width } = useWindowDimensions();
  const { route, descriptor } = scene;

  return (
    <HeaderContainer>
      <Svg height={135} width={width}>
        <Circle r={width} cx={width / 2} cy={-310} fill={COLOR.PRIMARY} />
      </Svg>
      <HeaderContentContainer>
        <HeaderContent>
          <If condition={Boolean(route.params?.title)}>
            <Title>{route.params.title}</Title>
            <Else />
            <Title>{descriptor.options.title}</Title>
          </If>
        </HeaderContent>
      </HeaderContentContainer>
    </HeaderContainer>
  );
};

Header.propTypes = {
  scene: PropTypes.shape({
    descriptor: PropTypes.shape({
      options: PropTypes.shape({
        title: PropTypes.string,
      }),
    }),
    route: PropTypes.shape({
      params: PropTypes.shape({
        title: PropTypes.string,
      }),
    }),
  }).isRequired,
};

export default Header;
