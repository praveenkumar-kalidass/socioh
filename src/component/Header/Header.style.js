import styled from 'styled-components/native';
import { Text } from 'native-base';

import { COLOR, FONT, SPACING } from '../../constant';

export const HeaderContainer = styled.SafeAreaView`
  background-color: ${COLOR.BACKGROUND};
`;

export const HeaderContentContainer = styled.View`
  position: absolute;
  padding-vertical: ${SPACING.SPACE_50};
  left: ${SPACING.SPACE_0};
  right: ${SPACING.SPACE_0};
`;

export const HeaderContent = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: space-around;
  padding-top: ${SPACING.SPACE_20};
`;

export const Title = styled(Text)`
  color: ${COLOR.WHITE};
  font-size: ${FONT.SIZE_24};
  font-weight: bold;
`;
