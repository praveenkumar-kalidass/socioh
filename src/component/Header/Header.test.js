import React from 'react';
import { render } from '@testing-library/react-native';

import Header from './index';

describe('Header', () => {
  const mockScene = {
    descriptor: {
      options: {
        title: 'Header',
      },
    },
    route: {},
  };

  it('should match snapshot', () => {
    const container = render(<Header scene={mockScene} />);

    expect(container).toMatchSnapshot();
  });

  it('should render title from params, if present', () => {
    const mockSceneWithRoute = {
      route: {
        params: {
          title: 'Praveen',
        },
      },
    };

    const { getByText } = render(<Header scene={mockSceneWithRoute} />);

    expect(getByText('Praveen')).toBeDefined();
  });
});
