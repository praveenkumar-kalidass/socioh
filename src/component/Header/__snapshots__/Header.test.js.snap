// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f0f8f8",
      },
    ]
  }
>
  <RNSVGSvgView
    bbHeight={135}
    bbWidth={750}
    focusable={false}
    height={135}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 135,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup>
      <RNSVGCircle
        cx={375}
        cy={-310}
        fill={4279943103}
        propList={
          Array [
            "fill",
          ]
        }
        r={750}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "paddingVertical": 50,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-around",
            "paddingTop": 20,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Header
      </Text>
    </View>
  </View>
</View>
`;
